core/src/main/java/org/apache/carbondata/core/memory/UnsafeMemoryManager.java

```diff
@@ -182,10 +224,15 @@ public long getUsableMemory() {
    */
   public static MemoryBlock allocateMemoryWithRetry(long taskId, long size)
       throws MemoryException {
+    return allocateMemoryWithRetry(taskId, size, false);
+  }
+
+  public static MemoryBlock allocateMemoryWithRetry(long taskId, long size, boolean isDriver)
```

Comment:
Better add overloaded method to take enum for offheap or onheap memory required.

Reply:
ok done